### PR TITLE
1452075: print only readable SSL error to console

### PR DESCRIPTION
**Pull request by jirihnidek**
* Bug fix: https://bugzilla.redhat.com/show_bug.cgi?id=1452075
* Print only readable part of SSL error to console
* When handling exception in register --force command, then do
  map_message to have nice error message.

### CODE REVIEW

**Comment by jirihnidek** [comment]
[Bug 1452075 - register --force option when provided with --serverurl option throws [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:579)](https://trello.com/c/cdUwZUhs/418-bug-1452075-register-force-option-when-provided-with-serverurl-option-throws-ssl-sslv3-alert-certificate-unknown-sslv3-alert-cer)

File: src/subscription_manager/exceptions.py [hunk 1]
```diff
@@ -74,7 +75,14 @@ def format_bad_ca_cert_exception(self, bad_ca_cert_error, message_template):
         return message_template % bad_ca_cert_error.cert_path
 
     def format_ssl_error(self, ssl_error, message_template):
-        return message_template % str(ssl_error)
+        # SSL error has fixed form: [SSL/BIO/...: ERRORO_ID] readable string (file.c:#line)
```
Comment (kahowell) [inline src/subscription_manager/exceptions.py]: Please revert the change to this method. It's okay to show the user the full error message, as long as we give them a human-readable message too.

File: src/subscription_manager/managercli.py [hunk 2]
```diff
@@ -1140,9 +1141,8 @@ def _do_command(self):
             except ssl.SSLError as e:
                 # since the user can override serverurl for register, a common use case is to try to switch servers
                 # using register --force... However, this normally cannot successfully unregister since the servers
-                # are different... So, do not map the message as SSLError is more useful than its mapped message in
-                # debugging this use case (mapped message is misleading in this use case)
-                handle_exception("Unregister failed: %s", e, map_message=False)
+                # are different.
+                handle_exception("Unregister failed: %s", e, map_message=True)
```
Comment (kahowell) [inline src/subscription_manager/managercli.py]: With this change, there are no instances of handle_exception called with `map_message=False`, so please remove the `map_message` logic introduced in 217c3863.

**Comment by jirihnidek** [comment]
When I will do both changes you requested, then there will not left anything in this PR :-). Shouldn't we close BZ report as not a bug then?

**Comment by kahowell** [comment]
@jirihnidek, There will be a change, it will just a few lines though:
- [line 168](https://github.com/candlepin/subscription-manager/blob/7f362a13be18dc6675a80180391795c136dbfd98/src/subscription_manager/managercli.py#L168) - remove map_message argument, also remove the related if statement
- [line 1147](https://github.com/candlepin/subscription-manager/blob/7f362a13be18dc6675a80180391795c136dbfd98/src/subscription_manager/managercli.py#L1147) - remove map_message argument

and then verify that the result looks sane.

File: src/subscription_manager/exceptions.py (hunk 1 above)
Comment (jirihnidek) [inline src/subscription_manager/exceptions.py]: We write more details to `rhsm.log` in `handle_exception()`. I vote for don't reverting this part :-). I thing main concern of bug reporter was style of error message.

**Comment by jirihnidek** [comment]
Let me, know if the last commit is OK. I will squash commits, when there will be no request on change.

File: src/subscription_manager/exceptions.py (hunk 1 above)
Comment (kahowell) [inline src/subscription_manager/exceptions.py]: This is hacky, imagine what happens if OpenSSL re-formats its error messages, for example, if they were to emit something like `[SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown [subject=CN=example.com] (_ssl.c:579)`, then the regex produces the empty string. We can't really predict how SSL handling might change. Already we are at the mercy of two libraries: python standard library `ssl` and `m2crypto`. Let's not couple ourselves to any particular SSL implementation anymore than we have to.

Comment (jirihnidek) [inline src/subscription_manager/exceptions.py]: You are right. Regular expressions are hungry. It can be solved using modification of regular expression to something like this: `output_error = re.sub('^\[[^\]]*\]|\([^\)]*\)$', '', str(ssl_error))`, but when I see it, I start to be little bit scared of this :-). Golden rule: do not use regular expression, then it becomes too complicated. Thus I will revert it.

BTW: OpenSSL is very conservative in changing error messages.

**Comment by jirihnidek** [comment]
@kahowell I reverted polishing of OpenSSL error. Let me know if the PR is OK. I will squash  commits then.

**Comment by kahowell** [comment]
@jirihnidek looks good. Please squash.

**Comment by jirihnidek** [comment]
Squashed.
